jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt

```diff
@@ -3,26 +3,293 @@
+                if (previousProfile != it.value) {
+                    profilesModified.add(ProfileCredentialsIdentifier(it.key))
+                }
+            }
+        }
+
+        // Any remaining profiles must have either become invalid or removed from the cred/config files
+        previousProfilesSnapshot.keys.asSequence().map { ProfileCredentialsIdentifier(it) }.toCollection(profilesRemoved)
+
+        profileHolder.update(newProfiles.validProfiles)
+        credentialLoadCallback(CredentialIdentifierChange(profilesAdded, profilesModified, profilesRemoved))
+
+        notifyUserOfResult(newProfiles, initialLoad)
+    }
+
+    private fun notifyUserOfLoadFailure(e: Exception) {
+        val loadingFailureMessage = message("credentials.profile.failed_load")
+
+        val detail = e.message?.let {
+            ": $it"
+        } ?: ""
+
+        notifyError(
+            title = message("credentials.profile.refresh_ok_title"),
+            content = "$loadingFailureMessage$detail",
+            action = createNotificationExpiringAction(ActionManager.getInstance().getAction("aws.settings.upsertCredentials"))
+        )
+    }
+
+    private fun notifyUserOfResult(newProfiles: Profiles, initialLoad: Boolean) {
+        val refreshTitle = message("credentials.profile.refresh_ok_title")
+        val totalProfiles = newProfiles.validProfiles.size + newProfiles.invalidProfiles.size
+        val refreshBaseMessage = message("credentials.profile.refresh_ok_message", totalProfiles)
+
+        // All provides were valid
+        if (newProfiles.invalidProfiles.isEmpty()) {
+            TelemetryService.recordSimpleTelemetry(null, "aws_credentials_load", true)
+
+            // Don't report we load creds on start to avoid spam
+            if (!initialLoad) {
+                notifyInfo(
+                    title = message("credentials.profile.refresh_ok_title"),
+                    content = refreshBaseMessage
+                )
+
+                return
+            }
+        }
+
+        // Some profiles failed to load
+        if (newProfiles.invalidProfiles.isNotEmpty()) {
+            val message = newProfiles.invalidProfiles.values.joinToString("\n") { it.message ?: it::class.java.name }
+
+            val errorDialogTitle = message("credentials.invalid.title")
+            val numErrorMessage = message("credentials.profile.refresh_errors", newProfiles.invalidProfiles.size)
+
+            notifyInfo(
+                title = refreshTitle,
+                content = "$refreshBaseMessage $numErrorMessage",
+                notificationActions = listOf(
+                    createShowMoreInfoDialogAction(message("credentials.invalid.more_info"), errorDialogTitle, numErrorMessage, message),
+                    createNotificationExpiringAction(ActionManager.getInstance().getAction("aws.settings.upsertCredentials"))
+                )
+            )
+
+            TelemetryService.recordSimpleTelemetry(null, "aws_credentials_load", false)
```

Comment:
should we stick this in the telemetry repo?

Reply:
Its conflicting. I think @hunterwerlla removed these

Reply:
I did, I think this PR should probably merge first though so there doesn't have to be as much rework. I can clean it up in my PR after this merges.